packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -2830,9 +2830,19 @@ abstract class _HasCreationLocation {
   _Location get _location;
 }
 
+/// A public api to access basic file information.
+abstract class SourceLocation {
+  /// File path of the location.
```

Comment:
I expect we will soon change this to be a
'package:path' whenever possible so I would rather we don't document these implementation details. 
I would go the route of adding a describeLocation method instead that returns a DiagnosticsNode object.
Otherwise we get to send out a breaking change e-mail in a week when file becomes uri, etc.

Reply:
If you want to expose it, expose it without any properties for now as you aren't using any of the properties anyway. that way We can add uri when it is available at some point.

Reply:
avoided exposing the class.